kubeflow/hub/api/model_registry_client.py

```diff
@@ -0,0 +1,288 @@
+
+        pip install 'kubeflow[hub]'
+
+    """
+
+    def __init__(
+        self,
+        base_url: str,
+        port: int | None = None,
+        *,
+        author: str | None = None,
+        is_secure: bool | None = None,
+        user_token: str | None = None,
+        custom_ca: str | None = None,
+    ):
+        """Initialize the ModelRegistryClient.
+
+        Args:
+            base_url: Base URL of the model registry server including scheme.
+                     Examples: "https://registry.example.com", "http://localhost"
+                     The scheme is used to infer is_secure and port if not explicitly provided.
+
+        Keyword Args:
+            port: Server port. If not provided, inferred from base_url scheme:
+                 - https:// defaults to 443
+                 - http:// defaults to 8080
+                 - no scheme defaults to 443
+            author: Name of the author.
+            is_secure: Whether to use a secure connection. If not provided, inferred from base_url:
+                      - https:// sets is_secure=True
+                      - http:// sets is_secure=False
+                      - no scheme defaults to True
+            user_token: The PEM-encoded user token as a string.
+            custom_ca: Path to the PEM-encoded root certificates as a string.
+
+        Raises:
+            ImportError: If model-registry is not installed.
+        """
+        try:
+            from model_registry import ModelRegistry
+        except ImportError as e:
+            raise ImportError(
+                "model-registry is not installed. Install it with:\n\n"  # fmt: skip
+                "  pip install 'kubeflow[hub]'\n"
+            ) from e
+
+        is_http = base_url.startswith("http://")
+        if is_secure is None:
+            is_secure = not is_http
+        if port is None:
+            port = 8080 if is_http else 443
+
+        self._registry = ModelRegistry(
+            server_address=base_url,
+            port=port,
+            author=author,  # type: ignore[arg-type]
+            is_secure=is_secure,
+            user_token=user_token,
+            custom_ca=custom_ca,
+        )
+
+    def register_model(
+        self,
+        name: str,
+        uri: str,
+        *,
+        version: str,
+        model_format_name: str | None = None,
+        model_format_version: str | None = None,
+        storage_key: str | None = None,
+        storage_path: str | None = None,
+        service_account_name: str | None = None,
+        model_source_kind: str | None = None,
+        model_source_class: str | None = None,
+        model_source_group: str | None = None,
+        model_source_id: str | None = None,
+        model_source_name: str | None = None,
+        author: str | None = None,
+        owner: str | None = None,
+        description: str | None = None,
+        metadata: Mapping[str, SupportedTypes] | None = None,
+    ) -> RegisteredModel:
```

Comment:
I've give `model_format_*` args default values and move `version` up. `name, uri, version` are the only required args.

@tarilabs Can we consider any of these args non-critical and remove them? It can be overwhelming to have so many args (i.e. `uri` plus also `storage_key` and `storage_path`). I see we have some info [here](https://github.com/kubeflow/model-registry/blob/dbf79f7c6763ccc5d31341fa38fc8e36767180c9/docs/logical_model.md?plain=1#L47-L63)

It's probably okay as long as we introduce it with simple examples.

Reply:
SGTM, thanks a lot Jon